### PR TITLE
New site redesign

### PR DESCRIPTION
**Pull request by aajjbb**
As I created the issue regarding the site in the awesome repo, I believe that the discussion fits better in the site repo itself.

This pull request has a re-designed menu for the site. It has colors similar to the ones used in the current docs page and is more 'minimalist'. 

@blueyed commented that the menu was too minimalist in the awesome-repo thread. Do you have any suggestions on how to make it better ?

### CODE REVIEW

File: local.css [hunk 1]
```diff
@@ -62,47 +64,46 @@ blockquote{
...
+    display: inline-block;
+    width: 100%;
+    margin-right: 0px;
+    float:right;
+    height: 50px;
 }
 
 #menu ul {
-	margin: 0;
+	/* padding-left: 20px; */
```
Comment (blueyed) [inline local.css]: Remove it?!

File: local.css [hunk 2]
```diff
@@ -7,20 +7,20 @@
...
+	width: 100%;	
+	color: black;
 	font-family: Verdana, Arial, sans-serif;
-        font-size: 0.8em;
+	font-size: 0.8em;
 }
 
 p {
-        text-align: justify;
+    text-align: justify;
```
Comment (blueyed) [inline local.css]: Unnecessary whitespace change?!

**Comment by blueyed** [comment]
> @blueyed commented that the menu was too minimalist in the awesome-repo thread. Do you have any suggestions on how to make it better ?

I don't remember what I've meant - do you have a link / more context, please?

The menu from this PR is simpler.
But the currently selected menu entry is not highlighted, is it?

**Comment by blueyed** [comment]
Ah, you mean @psychon's comment probably: https://github.com/awesomeWM/awesome/issues/1327#issuecomment-269898498.

**Comment by blueyed** [comment]
Current screenshot:
screenshot at 2016-12-31 18-57-50

**Comment by aajjbb** [comment]
Oh, sorry, I mistyped @blueyed instead of @psychon

File: local.css (hunk 2 above)
Comment (aajjbb) [inline local.css]: Fixed, the default whitespace is different from the one my emacs config uses

File: local.css (hunk 1 above)
Comment (aajjbb) [inline local.css]: Fixed

**Comment by aajjbb** [comment]
> The menu from this PR is simpler.

Yes, it is simpler, @psychon mentioned that it was too minimalist, so I asked for opinions on how to make it better.

> But the currently selected menu entry is not highlighted, is it?

With a mouse-hover it gets highlighted to black (the font color only)

**Comment by blueyed** [comment]
> With a mouse-hover it gets highlighted to black (the font color only)

Oh.. I've thought it was different currently already, so that the selected page/entry is highlighted also without mouse-over.
Can this be done with just CSS? - or would we need some ikiwiki (template) changes?!

**Comment by psychon** [comment]
> Can this be done with just CSS? - or would we need some ikiwiki (template) changes?!

No and no idea how to do it. ikiwiki does not really know that these pages are "special" to us. All that google found so far is that in a "PageSpec" the dot (".") refers to the current page. However, no idea how to use a PageSpec for anything.

**Comment by aajjbb** [comment]
It's easy when the tabs are on a single page, but as each tab is a different page, Javascript is the answer there.

**Comment by aajjbb** [comment]
Just implemented this 'highlight active tab' feature.

**Comment by psychon** [comment]
Took me an awful long time, but here is a CSS-only solution, I think:
```diff
diff --git a/local.css b/local.css
index 4528dd7..5502bac 100644
--- a/local.css
+++ b/local.css
@@ -136,6 +136,13 @@ blockquote{
        color: #fff;
 }
 
+ul.active-menu-about li.menu-about,
+ul.active-menu-download li.menu-download,
+ul.active-menu-doc li.menu-doc,
+ul.active-menu-community li.menu-community {
+       text-decoration: underline;
+}
+
 
 /**** Content Container ***/
 
diff --git a/templates/page.tmpl b/templates/page.tmpl
index 4d52033..7b616f4 100644
--- a/templates/page.tmpl
+++ b/templates/page.tmpl
@@ -19,12 +19,12 @@
             <TMPL_VAR SEARCHFORM>
         </TMPL_IF>
         <div id="menu">
-            <ul>
-                <li><a href="<TMPL_VAR BASEURL>index.html">home</a></li>
-                <li><a href="<TMPL_VAR BASEURL>download/">download</a></li>
-                <li><a href="<TMPL_VAR BASEURL>community/">community</a></li>
+            <ul class="active-menu-<TMPL_VAR NAME="title">">
+                <li class="menu-about"><a href="<TMPL_VAR BASEURL>index.html">home</a></li>
+                <li class="menu-download"><a href="<TMPL_VAR BASEURL>download/">download</a></li>
+                <li class="menu-community"><a href="<TMPL_VAR BASEURL>community/">community</a></li>
                 <li><a href="<TMPL_VAR BASEURL>apidoc/">doc (git/master)</a></li>
-                <li><a href="<TMPL_VAR BASEURL>doc/">doc (v4.0)</a></li>
+                <li class="menu-doc"><a href="<TMPL_VAR BASEURL>doc/">doc (v4.0)</a></li>
                 <li><a href="https://github.com/awesomeWM/awesome/issues">bugs/issues</a></li>
             </ul>
         </div>
```
This will produce broken HTML if the page name (I think that's the base name of the file) contains "funny" characters, but I am ok with assuming that this does not happen. :-)

**Comment by aajjbb** [comment]
Which solution to choose ? Pure CSS or JS ?

**Comment by psychon** [comment]
What would be better about a JS solution? I thought less JS is always better...?

**Comment by aajjbb** [comment]
I agree. I'll merge this CSS only solution.

File: community.mdwn [hunk 3]
```diff
@@ -1,6 +1,4 @@
-<h1 id="page-id">
-    Community
-</h1>
+#Community
```
Comment (blueyed) [inline community.mdwn]: Should have a space after the `#` I think.. (also below)

File: download.mdwn [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Download
+#Download
```
Comment (blueyed) [inline download.mdwn]: Should keep the space here (also for other (section) headers).

Comment (aajjbb) [inline download.mdwn]: done

**Comment by aajjbb** [comment]
More suggestion on how to improve the site ?

**Comment by Elv13** [comment]
I think it would be a good idea to have a template for user screenshots. Like everybody is allowed a "bigger" screenshot (300-something per 200-something), (up to) 4 smaller thumbs (with big screenshot when you click on them), a 500 char description and a [GitHub] link. 

Configs like mine, @mindeunix, @actionless and @copycat-killer are too customized to be represented by a single screenshot. Of course, those are the exteremes. Most users prefer more minimalist configs, but what's the point of having screenshot if they don't display the full capability of the software.

**Comment by aajjbb** [comment]
In my opinion, those more elaborated setups should be in a 'recipes' tab, where we should post useful snippets, detailed setups and important legacy stuff from the old wiki, it would be cleaner to keep screenshots tab with a few examples of what awesome is capable of.

**Comment by copycat-killer** [comment]
I agree. We need the wiki back with a *showcase* section. It'll not only show the WM capabilities, but also some UI design tips.

**Comment by psychon** [comment]
I agree with "more examples", but this pull request is about a new design. Let's try to get this back on track:

Assuming this PR is considered mergeable (I guess I would clean up the commit history a bit, but that can still be done later...): Could we get a new comment with screenshots (perhaps also of the old site, for comparison) of what this looks like? Then, **everyone** can say "the new design is better" / "I am neutral" / "Nooo! Stay with the current desgin".

**Comment by aajjbb** [comment]
Here are some screenshots of the 'new' site. I'm sorry for the 'dirt' commit issue. 

screenshot from 2017-01-06 16-57-50
screenshot from 2017-01-06 16-57-55
screenshot from 2017-01-06 16-57-58
screenshot from 2017-01-06 16-58-03

**Comment by aajjbb** [comment]
Here are screenshots from the current site.
screenshot from 2017-01-06 17-08-24
screenshot from 2017-01-06 17-08-28
screenshot from 2017-01-06 17-08-31

**Comment by aajjbb** [comment]
Votes here, please. :
New Site: :+1: 
Old Site :heart:

**Comment by blueyed** [comment]
What about using a logo instead of the name in the upper left?
autogenerated apidoc logo

Is this considered to be the official logo after all (it's from the apidocs)?

**Comment by aajjbb** [comment]
Using the Awesome logo. Opinions ?

screenshot from 2017-01-07 00-41-21

**Comment by blueyed** [comment]
> Using the Awesome logo. Opinions ?

The text should start next to it - or rather: the logo should not push the content down so much.

**Comment by blueyed** [comment]
Would be nice to have the header "Community" lined up with "home" from the menu.

**Comment by aajjbb** [comment]
Sure, I just posted to give a perspective of how it would look like

**Comment by Elv13** [comment]
For the screenshot, I propose adding my SciFi one and use the other "material" screenshot instead of the one from the front page (from the same author). Both have the correct copyright re-distribution rights.

http://imgur.com/I7eL5it (the alternate version of the frontpage one)
http://imgur.com/a/thZiW (first one)

As for the logo, maybe having the horizontal "AWESOME" (technically the official alternate logo) block-font instead of the big "A"+vertical "AWESOME" would look a little out of place. I preferred the earlier version "awesome" name size. The CI currently doesn't generate an official horizontal SVG, it will require a patch to re-use @actionless code for it.

**Comment by Elv13** [comment]
> Is this considered to be the official logo after all (it's from the apidocs)?

It's in theory the second alternate logo. The "official" logo is the square "A", the alternate "AWESOME" horizontally. This one has been seen before, but was never really official (I think I came up with it first). I like it and think it looks great in situations where the default logo looks a little bland. If your question was about the exact aspect ratio of the "tiles" within the letters: I don't know. It looks right to me.

**Comment by aajjbb** [comment]
I liked the horizontal logo @copycat-killer suggested

**Comment by copycat-killer** [comment]
awesome_logo_horizontal

[Here](https://linx.li/selif/4dw3xqas.svg) is a vector version.

**Comment by aajjbb** [comment]
Using the horizontal icon. I also kept the icon on left and made the menu options flow right. I do think the icon is a 'little' to bigger but I don't look prettier when it gets smaller due to small letter spacing.
screenshot from 2017-01-08 16-30-09

**Comment by copycat-killer** [comment]
Doubled letter spacing.

awesome_logo_horizontal2

[Vector](https://linx.li/9u2742i4.svg)

You can add more space yourself, if you want: just open the vector image in Inkscape and move the letter objects.

**Comment by aajjbb** [comment]
Here with double space:
screenshot from 2017-01-08 21-49-54

**Comment by blueyed** [comment]
Using this logo I would suggest for it to be at the top, and the page/menu centered below it?!

**Comment by aajjbb** [comment]
Those screenshots are taken from inside a VirtualBox machine that has a pretty low resolution (ikiwiki doesn't build in Arch/Manjaro because of broken perl dependencies). It would be nice if someone with good resolution took the screenshots.

 I kind of thought about letting the 'big logo' floated to the left side of the page and the main-menu items to float to the right, any thoughts on that idea ?

**Comment by copycat-killer** [comment]
With the doubled spacing, you should be able to reduce the horizontal logo size more, like in my original screenshot.

**Comment by copycat-killer** [comment]
>  It would be nice if someone with good resolution took the screenshots.

On my 1080p monitor.

**Comment by aajjbb** [comment]
Resized again. Now it's smaller in width but a little bigger in height as it seems to the flatted in greater resolutions

**Comment by copycat-killer** [comment]
What about this?



```diff
--- a/local.css	2017-01-09 17:35:46.206296384 +0100
+++ b/local.css	2017-01-09 17:35:46.316293832 +0100
@@ -98,39 +98,33 @@
 #menu {
     display: inline-block;
     width: 100%;
-    margin-right: 0px;
-    float:right;
+    float:center;
     height: 110px;
 }
 
 #menu img {
-        float: left;
-        /*! padding-right: 10px; */
-        padding-top: 10px;
-        height: 38%;
-        width: 23%;
-        margin-left: 20px;
+    /*! padding-right: 10px; */
+    padding-top: 20px;
+    width: 20%;
+    margin-left: 40%;
 }
 
 #menu ul {
-        float: right;
-}
-
-#menu ul {
-	list-style: none;
+    overflow: auto;
+    list-style: none;
+    margin-left: 30%;
 }
 
 #menu li {
 	display: block;
 	float: left;
-	white-space: nowrap;
 }
```

```diff
--- a/index.mdwn	2017-01-09 17:45:13.209810337 +0100
+++ b/index.mdwn	2017-01-09 17:35:42.203055858 +0100
@@ -1,7 +1,3 @@
-<div id="logo">
-[[!img images/icons/aw_64_shadow.png alt="awesome logo"]]
-</div>
-
 # Home
 
 **awesome** is a highly configurable, next generation framework window
```

Don't use this patch as is, but to find a proper way to center stuff in CSS for every screen width.

File: local.css [hunk 5]
```diff
@@ -91,29 +91,29 @@ blockquote{
 	color: black;        
 }
 
+/**** Center Logo ****/
+
+#center-logo img {
+        padding-top: 20px;
+        width: 20%;
+        margin-left: 40%;
```
Comment (blueyed) [inline local.css]: Does this scale to (very) low resolutions?!

**Comment by Elv13** [comment]
Can we add the ribbons now that the right corner is free?

https://github.com/blog/273-github-ribbons (the red one?)

**Comment by aajjbb** [comment]
Habemus Ribbon

File: local.css (hunk 5 above)
Comment (aajjbb) [inline local.css]: No issues in my 'tests'. The image is a svg so there aren't much issues.

File: local.css [hunk 6]
```diff
@@ -151,7 +163,7 @@ blockquote{
 }
 
 #content h2 {
-	margin: 0;
+	margin: 0;
```
Comment (blueyed) [inline local.css]: Trailing whitespace.

File: local.css [hunk 7]
```diff
@@ -46,94 +51,101 @@ blockquote{
...
 	text-decoration: none;
 }
 
 #top h2 a:hover {
-        color: #347272;
+	color: black;        
+}
+
+/**** Center Logo ****/
+
```
Comment (blueyed) [inline local.css]: Not sure about the "header" itself, but it should at least have no newline below.
What about Vim markers.. '{{{` … `}}}`

File: local.css [hunk 8]
```diff
@@ -46,94 +51,101 @@ blockquote{
...
 #menu {
-	display: block;
-	float:right;
+    display: inline-block;
+    width: 100%;
+    height: 110px;
+}
+
+#menu ul {    
+        overflow: auto;
```
Comment (blueyed) [inline local.css]: indent is off from below.

File: local.css [hunk 9]
```diff
@@ -46,94 +51,101 @@ blockquote{
...
 	letter-spacing: 1px;
-	color: gray;
-	background: url(images/bg_menu.gif);
+	color: gray;	
 }
 
-#menu li a.current:hover {
-	color: #fff;
+#menu li a.current:hover {	
 }
```
Comment (blueyed) [inline local.css]: Empty by now, so should be removed.

File: local.css [hunk 10]
```diff
@@ -172,10 +188,10 @@ blockquote{
...
 
 #left {
-	width: 700px;
+	width: 60%;
 	float:left;
 	display: block;
-        margin-left: 20px;
+    margin-left: 20%;
```
Comment (blueyed) [inline local.css]: Indent is off.

File: local.css [hunk 11]
```diff
@@ -279,9 +293,9 @@ blockquote{
 }
 
 #donate {
-  margin: 1em 0;
-  text-align: center;
+        margin: 1em 0;
+	text-align: center;
```
Comment (blueyed) [inline local.css]: Indent does not match.

File: local.css [hunk 12]
```diff
@@ -279,9 +293,9 @@ blockquote{
...
 }
 #donate p, #donate div {
-  text-align: center;
+        text-align: center;
```
Comment (blueyed) [inline local.css]: Indent is not 4, is it?

Comment (aajjbb) [inline local.css]: It's 8. Fixed.

File: local.css (hunk 10 above)
Comment (aajjbb) [inline local.css]: Fixed.

File: local.css (hunk 6 above)
Comment (aajjbb) [inline local.css]: Fixed.

File: local.css (hunk 9 above)
Comment (aajjbb) [inline local.css]: Fixed

File: local.css (hunk 7 above)
Comment (aajjbb) [inline local.css]: I do use emacs so I'm not aware of Vim markers.

File: local.css (hunk 8 above)
Comment (aajjbb) [inline local.css]: Fixed.

**Comment by blueyed** [comment]
Looks good!

I think we do not `#logo` anymore:

```diff
diff --git i/index.mdwn w/index.mdwn
index 49e6f15..d851510 100644
--- i/index.mdwn
+++ w/index.mdwn
@@ -1,7 +1,3 @@
-<div id="logo">
-[[!img images/icons/aw_64_shadow.png alt="awesome logo"]]
-</div>
-
 # Home
 
 **awesome** is a highly configurable, next generation framework window
diff --git i/local.css w/local.css
index f9823ae..ed6703b 100644
--- i/local.css
+++ w/local.css
@@ -165,7 +165,6 @@ blockquote{
 
 /**** Content ***/
 
-#logo { float: right; margin: 32px 32px 32px 32px; }
 #screenshot { text-align: center; margin: 32px 32px 32px 32px; }
 #screenshot img { border: 2px solid #347272; }
 
```

File: local.css (hunk 7 above)
Comment (blueyed) [inline local.css]: It's not just Vim, but also jEdit at least.. ;)
Just a remark anyway.

**Comment by blueyed** [comment]
Can you also please link to the `index.html` files explicitly?! - makes it easier to browse it without an actual HTTP server-

**Comment by aajjbb** [comment]
The way ikiwiki builds the site, each page goes to a proper folder with an `index.html` page, which make relative paths unfeasible (the path from a page when you are in `home` is different from when you are in `download`. I don't know an elegant way to fix it.

**Comment by psychon** [comment]
@blueyed https://ikiwiki.info/tips/JavaScript_to_add_index.html_to_file:_links/

> which make relative paths unfeasible 

At least the template does use relative paths: https://github.com/awesomeWM/awesome-www/blob/40598b5e42fa1ab354b9096b68b7736f7a737f0d/templates/page.tmpl#L23

**Comment by aajjbb** [comment]
The script isn't perfect but works, at least I don't need nginx to proper test the changes anymore.

**Comment by psychon** [comment]
(The past few times that I worked on the webpage, I just lived with the links to directories and just clicked on index.html manually.)

**Comment by blueyed** [comment]
@aajjbb 
`python3 -m http.server -b localhost 8000` would also work.. added to the README in #57.

**Comment by blueyed** [comment]
@aajjbb 
It's probably better to not include the JS: it is only useful/required for local development.
Let's change the wording in #57 instead.

**Comment by aajjbb** [comment]
Just removed.

File: README.md [hunk 13]
```diff
@@ -9,6 +9,12 @@ The website is based on [ikiwiki](https://ikiwiki.info/). You can build the web
 page locally by running `make`. The result will be in `html/`. Besides ikiwiki,
 you will also need [PerlMagick](https://www.imagemagick.org/script/perl-magick.php).
 
+## Contribution Guide
+
+### Screenshots
+
+To contribute with Screenshots, please add your image to the folder images/screenshots and be sure to keep your image inside an <figure> tag and to fill the <figcaption> with what is being used on the screenshot.
```
Comment (blueyed) [inline README.md]: Should be wrapped to ~80 chars.

**Comment by aajjbb** [comment]
Now with https://github.com/awesomeWM/awesome-www/pull/56/commits/fa490ef0c8d15ca6ce6093dcfd53c18b04e8eaa5, the way we work with screenshots is improved.

File: README.md (hunk 13 above)
Comment (blueyed) [inline README.md]: It should also mention to add it to `screenshots.mdwn`?!

**Comment by blueyed** [comment]
A lot of (constructive) bike shedding here since 11 days.. ;)

**Comment by blueyed** [comment]
> Now with fa490ef, the way we work with screenshots is improved.

This still does not scale, but is good enough for now / this PR.

File: README.md (hunk 13 above)
Comment (aajjbb) [inline README.md]: Fixed. I improved the language and explaining the process through topics.

**Comment by aajjbb** [comment]
It doesn't scale because of size issues with images ?

**Comment by aajjbb** [comment]
Did my first interactive rebase ever on this pull request. Hope the history is cleaner and better.

**Comment by blueyed** [comment]
> It doesn't scale because of size issues with images ?

No, more like with a lot of submissions in the future.
It might be better to run some gallery (plugin) by then.

**Comment by blueyed** [comment]
@aajjbb 
Please try the debase again: this PR/branch should not include things already in master.
Make sure to fetch master before debasing.

**Comment by aajjbb** [comment]
I think I made it, it wasn't easy for a first times.

**Comment by blueyed** [comment]
There is some conflict in the local.css still.

**Comment by aajjbb** [comment]
Just resolved the conflict from the github interface. I won't mention people in comment again.

**Comment by blueyed** [comment]
I think this is good to go.
What do others think?

Before merging the commits should be squashed / cleaned up, especially the ones that have been reverted again (JS snippet).

**Comment by mindeunix** [comment]
New design looks really nice, but have a few HTML warnings/errors like:
```html
Wrong: <img src="image.png" alt="image">
Ok:    <img src="image.png" alt="image" />
```
http://validator.w3.org/check

**Comment by aajjbb** [comment]
I'll validate the html in this checker and update.

**Comment by aajjbb** [comment]
Squashed them to 12 commits.

**Comment by aajjbb** [comment]
Solved the necessary errors pointed by the checker. As I searched, some of the are obsolete and I didn't messed with them (Like adding xmlns to the page).

**Comment by blueyed** [comment]
Let's keep awesomeWM awesome still.. :)
Thanks a lot

**Comment by aajjbb** [comment]
I'm so happy to help :+1:
